File: 2840-check-if-strings-can-be-made-equal-with-operations-ii/2840-check-if-strings-can-be-made-equal-with-operations-ii.cpp
class Solution {
public:
    bool checkStrings(string s1, string s2) {

        vector<int> odd(26, 0), even(26, 0);
        
        for (int i = 0; i < s1.size(); i++) {
            char c = s1[i];
            if (i % 2 == 0) {
                even[c - 'a']++;
            } else {
                odd[c - 'a']++;
            }
        }

        for (int i = 0; i < s2.size(); i++) {
            char c = s2[i];
            if (i % 2 == 0){
                if (even[c - 'a'] == 0)
                    return false;
                even[c - 'a']--;
            } else {
                if (odd[c - 'a'] == 0)
                    return false;
                odd[c - 'a']--;
            }
        }

        return true; 
    }
};

File: 2840-check-if-strings-can-be-made-equal-with-operations-ii/2840-check-if-strings-can-be-made-equal-with-operations-ii.py
from collections import Counter

class Solution:
    def checkStrings(self, s1: str, s2: str) -> bool:
        
        odd_s1, even_s1, odd_s2, even_s2 = [], [], [], []

        for i, c in enumerate(s1):
            if i % 2 == 0:
                even_s1.append(c)
            else:
                odd_s1.append(c)
        
        for i, c in enumerate(s2):
            if i % 2 == 0:
                even_s2.append(c)
            else:
                odd_s2.append(c)
        
        return Counter(odd_s1) == Counter(odd_s2) and Counter(even_s1) == Counter(even_s2)